app/controllers/concerns/jwt_authorization_concern.rb

```diff
@@ -0,0 +1,62 @@
+# Inspects the HTTP Authorization header for a Bearer JWT token and
+# populates the @auth_token_payload instance variable with the JWT payload.
+# This does no signature verification of the JWT, expecting a gateway or
+# proxy to have done that already.
+# If the payload cannot be parsed or does not have a preferred_username
+# attribute, a 401 Unauthorized response with an empty JSON body will be
+# returned.
+# All private API endpoints should include this concern and can rely on the
+# preferred_username to represent the user.
+module JwtAuthorizationConcern
+  extend ActiveSupport::Concern
+
+  included do
+    before_action :check_token
+  end
+
+  private
+
+  def check_token
+    @auth_token_payload = nil
+    @current_user = nil
+
+    jwt = nil
+
+    auth_header = request.headers['Authorization']
+    if !auth_header.nil?
+      m = auth_header.match(/^Bearer (.*)$/)
+      if m && m.captures.length == 1
+        pieces = m.captures[0].split('.')
+        if pieces.length == 3
+          begin
+              jwt = JSON.parse(Base64.decode64(pieces[1]))
```

Comment:
Should we verify the JWT here?

Reply:
I noticed this is stacked on top of PR https://github.com/NetrunnerDB/netrunnerdb-api-server/pull/165. If you want to avoid those changes showing up here you could set that as the base branch.

Reply:
I'm not sure you can do that from a fork. I don't know another way to avoid seeing common commits on both PRs.

Reply:
i think i might need to set up a different base when creating the PR.  Anyway, problem for another PR.